source/is-equal.d.ts

```diff
@@ -0,0 +1,45 @@
+/**
+Returns a boolean for whether the two given types are equal.
+
+@link https://github.com/microsoft/TypeScript/issues/27024#issuecomment-421529650
+@link https://stackoverflow.com/questions/68961864/how-does-the-equals-work-in-typescript/68963796#68963796
+
+@example
+```
+import type {IsEqual} from 'type-fest';
+
+// `IsEqual` compares 1 and 1, then gets true as a result.
+type ExpectTrue = IsEqual<1, 1>;
+//=> true
+```
+
+@example
+```
+import type {IsEqual} from 'type-fest';
+
+// `IsEqual` compares 2 and 1, then gets false as a result.
+type ExpectFalse = IsEqual<2, 1>;
```

Comment:
I don't really see the point of this and the previous example. No one would actually do this.

Reply:
I thought these examples would help users understand, but you are right, these examples will not be used. Therefore, I have removed the examples.